### PR TITLE
2362 remove message sequence protection

### LINKED ISSUE #2362: Remove code ensuring that Kafka trigger honours message sequence for messages with keys
In the original implementation, the Kafka trigger  used `TriggerKafakMessage` records as a buffer between the incoming message and the generation of WorkOrder instances. These records ensured that the Kafka trigger would honour message sequence for messages with keys.

This is not a good fit wrt other areas of Lightning's design and so this functionality must be  removed and will be replaced, in the near future, by a higher-level construct that will generalise message sequence protection.

### PR DESCRIPTION
**Pull request by rorymckinley**
### Description

This PR removes any code that was used to specifically manage Kafka messages with keys, ensuring that the message sequence was preserved during processing.

As there is quite a lot going on I grouped the changes in distinct commits, in the hopes that it makes digestion easier.

Closes #2362 

### Validation steps

- Setup a local Kafka cluster based on these [instructions](https://github.com/OpenFn/lightning/blob/main/kafka_testing/KAFKA_ARCHITECTURE.md#testing-locally-using-the-docker-kafka-cluster).
- Ensure that `KAFKA_TRIGGERS_ENABLED` is set to 'yes'
- Set up a Kafka trigger similar t the screenshot below (assuming you have  a topic named `baz_topic`) :

Screenshot from 2024-08-07 10-07-28

- Publish a message using kcat: `cat message.json | kcat -P -b 127.0.0.1:9094 -t baz_topic`.
- You should see a WorkOrder being enqueued.

### Pre-submission checklist

- [x] I have performed a **self-review** of my code.
- [x] I have implemented and tested all related **authorization policies**. (e.g., `:owner`, `:admin`, `:editor`, `:viewer`)
- [x] I have updated the **changelog**.

### CODE REVIEW

**Review by jyeshe — CHANGES_REQUESTED** [review]
More than happy to review this going back to the original proposed design. It improved a lot the responsiveness and I think we just need to prioritize the rate limiting to protect an overload on the indexation of the workorders dataclips. @christad92 

It looks good for me and I would ask just a rename on an error message to make it consistent with the data format.

File: lib/lightning/kafka_triggers/message_handling.ex [hunk 1]
```diff
@@ -102,131 +21,40 @@ defmodule Lightning.KafkaTriggers.MessageHandling do
...
+    else
+      {:error, %Ecto.Changeset{} = error_changeset} ->
+        {:error, error_changeset}
+
+      {:error, %Jason.DecodeError{}} ->
         {:error, :data_is_not_json_object}
 
-      {:error, _decode_error} ->
+      false ->
         {:error, :data_is_not_json_object}
```
[the remark this answers is not in the document]
Comment (rorymckinley) [reply lib/lightning/kafka_triggers/message_handling.ex]: @jyeshe thanks - I updated it in a new commit.

**Review by jyeshe — APPROVED** [review]
🎉

**Review by christad92 — APPROVED** [review]
Thanks! amazing work!

**Pull request merged**